new-packages/ts-project/src/state.ts

```diff
@@ -124,6 +122,64 @@ function getObjectTransitionTargets(
   });
 }
 
+function extractActionBlocks(
+  ctx: ExtractionContext,
+  ts: typeof import('typescript'),
+  expression: Expression,
+  { parentId }: { parentId: string },
+) {
+  return mapMaybeArrayElements(
+    ts,
+    expression,
+    (element): ActionBlock | undefined => {
+      if (isUndefined(ts, element)) {
+        return;
+      }
+      if (ts.isStringLiteralLike(element)) {
+        return createActionBlock({
+          sourceId: element.text,
+          parentId,
+        });
+      }
+      if (ts.isObjectLiteralExpression(element)) {
+        const typeProperty = findProperty(ctx, ts, element, 'type');
+
+        if (!typeProperty) {
+          return;
+        }
+
+        if (ts.isStringLiteralLike(typeProperty.initializer)) {
+          return createActionBlock({
+            sourceId: typeProperty.initializer.text,
+            parentId,
+          });
+        }
+      }
+
+      return createActionBlock({
+        sourceId: `inline:${uniqueId()}`,
+        parentId,
+      });
```

Comment:
This means an invalid value for type property in a parameterized action will be extracted as an inline action. This is fine as long as it is our intention.

Reply:
I don't think that this is the case. Parameterized actions are extracted in the conditional block above this and when `typeProperty` is not found it `return`s

Reply:
I mean for a case like `action: {type: {}}`, type property will be found but it's not a string so the code continues and extract it as an inline action whereas this is invalid.

Reply:
Ah, I see what you mean - thanks for pointing this out. This should definitely be reported as an error.